Send PIPs to Archivematica as BagIt bags

djjuhasz opened the pull request:
Fixes #805

- Remove the bundle activity from the AM processing workflow
- Change the package type to "zipped bag" when starting a transfer via
  the Archivematica API
- Bag the PIP before sending it to Archivematica (if it's not already a
  bag)
- Add a "TransferSourcePath" config value to specify the API path to the
  Transfer Source directory where PIPs are uploaded

djjuhasz committed:
Send PIPs to Archivematica as BagIt bags

Fixes #805

- Move the bundle activity to the a3m branch of the processing workflow
- Change the package type to "zipped bag" when starting a transfer via
  the Archivematica API
- Bag the PIP before sending it to Archivematica (if it's not already a
  bag)
- Add a "TransferSourcePath" config value to specify the API path to the
  Transfer Source directory where PIPs are uploaded

jraddaoui commented on hack/kube/overlays/dev-am/enduro-am.yaml:
```diff
@@ -81,6 +81,11 @@ spec:
+              valueFrom:
+                secretKeyRef:
+                  name: enduro-am-secret
+                  key: transfer_source_path
```
If you are planning to include it in the secret, please, add it in the documentation:

https://github.com/artefactual-sdps/enduro/blob/main/docs/src/dev-manual/archivematica.md?plain=1#L78

I also think it's okay not to consider it in the secret, as it's done with some of the other config options.

djjuhasz replied:
Okay, I'll add it to the docs.  

I don't think it needs to be a secret, except that I want to be able to change the transfer source path in dev without modifying `enduro.toml` (or any other files that might get commited) and there's no other way to do that AFAIK.

jraddaoui commented on internal/workflow/processing.go:
```diff
@@ -803,51 +772,98 @@ func (w *ProcessingWorkflow) waitForReview(ctx temporalsdk_workflow.Context) *pa
+	}
+
+	// Zip PIP.
+	activityOpts := withActivityOptsForLongLivedRequest(ctx)
```
Should we use `withActivityOptsForLocalAction(ctx)` instead of `withActivityOptsForLongLivedRequest(ctx)` for the zip activity too?

djjuhasz replied:
Yeah, good idea. I can't imagine zipping a PIP to take longer than an hour. :sweat_smile:

jraddaoui commented on internal/workflow/processing_test.go:
```diff
@@ -663,21 +666,37 @@ func (s *ProcessingWorkflowTestSuite) TestAMWorkflow() {
+	// 	mock.Anything,
+	// 	mock.Anything,
+	// 	mock.Anything,
+	// ).Return(nil, nil).Never()
```
?

djjuhasz replied:
@jraddaoui oh yeah, I meant to delete that call.  That was at line 669 before I re-wrapped the lines — it wasn't commented, but as far as I can tell it doesn't do anything because of the `Never()` method at the end there. :shrug:

jraddaoui reviewed (changes requested):
Thanks @djjuhasz, this looks great!

Some minor comments below and a bigger request after seeing the changes from both PRs (more related to #999 than to this one).

We decided at some point that preprocessing child workflows MUST return a valid Bag. If a preprocessing child workflow has been executed and the package is not identified as a Bag, we should fail the workflow instead of making it a Bag.

Check the following diagram that @sallain wrote a while ago, it doesn't include the identify activity but it tries to represent that, if a child workflow was run, we must have a valid Bag.

https://github.com/artefactual-sdps/enduro/issues/805#issuecomment-2040656506

djjuhasz commented:
@jraddaoui okay, I've addressed your comments and the workflow will now stop if preprocessing returns a SIP that is not a bag.  I also had to fight with prettier a bit about formatting markdown, and made some changes to the preprocessing_test script to get the preprocessing test working and for my sanity trying to figure out where preservation tasks are being created and completed.

I'll squash all the changes before merging.

jraddaoui reviewed (approved):
Thanks @djjuhasz!

djjuhasz committed:
Add markdown files to prettier ignore list

After configuring prettier to use an indent of four spaces for Markdown
files it is adding a four space indent after the "-" character in
bulleted lists (see prettier/prettier#5019).

For now configure prettier to ignore Markdown files altogether to avoid
incorrect formatting changes.
